src/platform_impl/linux/x11/mod.rs

```diff
@@ -100,22 +100,27 @@ impl<T: 'static> EventLoop<T> {
             .expect("Failed to query XRandR extension");
 
         let xi2ext = unsafe {
-            let mut result = XExtension {
-                opcode: mem::uninitialized(),
-                first_event_id: mem::uninitialized(),
-                first_error_id: mem::uninitialized(),
-            };
+            let mut opcode = MaybeUninit::uninit();
```

Comment:
I think it should be safe to zero-initialize the whole struct, then pass pointers to the individual members in it.